src/Samples/SubModelSeq/Program.fs

```diff
@@ -13,181 +28,195 @@ module List =
 
   /// Returns the sibling counters of the specified counter ID.
   let childrenCountersOfParentOf cid =
-    parentOf cid >> Tree.dataOfChildren
-
-  type Msg =
-    | ToggleGlobalState
-    | AddCounter of parent: CounterId
-    | Increment of CounterId
-    | Decrement of CounterId
-    | SetStepSize of CounterId * int
-    | Reset of CounterId
-    | Remove of CounterId
-    | MoveUp of CounterId
-    | MoveDown of CounterId
-
-  /// Updates the counter using the specified function if the ID matches,
-  /// otherwise passes the counter through unmodified.
-  let updateCounter f id c =
-    if c.Id = id then f c else c
-
-  let incrementCounter = updateCounter Counter.increment
-  let decrementCounter = updateCounter Counter.decrement
-  let setStepSizeOfCounter ss = updateCounter <| Counter.setStepSize ss
-  let resetCounter = updateCounter Counter.reset
-
-  let update msg m =
-    match msg with
-    | ToggleGlobalState -> { m with SomeGlobalState = not m.SomeGlobalState }
-
-    | AddCounter pid ->
-        let f (n: Tree.Node<Counter>) =
-          if n.Data.Id = pid
-          then { n with Children = (Counter.create () |> Tree.asLeaf) :: n.Children}
-          else n
-        { m with DummyRoot = m.DummyRoot |> Tree.map f }
-
-    | Increment id -> { m with DummyRoot = m.DummyRoot |> Tree.mapData (incrementCounter id) }
-
-    | Decrement id -> { m with DummyRoot = m.DummyRoot |> Tree.mapData (decrementCounter id) }
-
-    | SetStepSize (id, ss) -> { m with DummyRoot = m.DummyRoot |> Tree.mapData (setStepSizeOfCounter ss id) }
-
-    | Reset id -> { m with DummyRoot = m.DummyRoot |> Tree.mapData (resetCounter id ) }
-
-    | Remove id ->
-        let f (n: Tree.Node<Counter>) =
-          { n with Children = n.Children |> List.filter (fun n -> n.Data.Id <> id) }
-        { m with DummyRoot = m.DummyRoot |> Tree.map f }
-
-    | MoveUp id ->
-      let f (n: Tree.Node<Counter>) =
-        match n.Children |> List.tryFindIndex (fun nn -> nn.Data.Id = id) with
-        | Some i -> { n with Children = n.Children |> List.swap i (i - 1) }
-        | None -> n
-      { m with DummyRoot = m.DummyRoot |> Tree.map f }
-
-    | MoveDown id ->
-      let f (n: Tree.Node<Counter>) =
-        match n.Children |> List.tryFindIndex (fun nn -> nn.Data.Id = id) with
-        | Some i -> { n with Children = n.Children |> List.swap i (i + 1) }
-        | None -> n
-      { m with DummyRoot = m.DummyRoot |> Tree.map f }
+    cid |> parentOf >> RoseTree.dataOfChildren
 
 
 module Bindings =
 
   open App
 
-  let rec counterTreeBindings () : Binding<Model * Counter, Msg> list = [
-    "CounterIdText" |> Binding.oneWay(fun (_, { Id = CounterId id }) -> id)
-
-    "CounterValue" |> Binding.oneWay(fun (_, c) -> c.CounterValue)
-
-    "Increment" |> Binding.cmd(fun (_, c) -> Increment c.Id)
-
-    "Decrement" |> Binding.cmd(fun (_, c) -> Decrement c.Id)
+  let rec counterTreeBindings () : Binding<Model * Identifiable<Counter>, Msg> list = [
```

Comment:
Any particular reason you have changed the spacing between the bindings in `counterTreeBindings`? I don't care that much, but I can't see any particular pattern.

Reply:
While maintaining the order, I remove the blank lines between the bindings specific to a single counter, which are the ones named `CounterValue`, `Increment`, `Decrement`, `StepSize`, and `Reset`.

When I started making the changes in this branch, my goal was not merely to simplify the code in the `SubModelSeq` sample.  My goal was to extract those five bindings.  So my first first change might have been to remove those blank lines.  All the other changes were either to reduce the "noise" to help me focus on my goal or to do something useful if I didn't know what else to do to achieve my goal.

In the end, removing those lines is not important for this PR.  If you want, I can put them back.

Reply:
No, it's fine as it is. It's moot point once #244 is merged.